### PR TITLE
[DATA-535] Create an orbslam integration test that runs mono YCbCr im…

### PR DESCRIPTION
**Pull request by tessavitabile**
…ages

### CODE REVIEW

File: rimage/image_processing.go [hunk 1]
```diff
@@ -508,3 +508,51 @@ func EdgeHysteresisFiltering(mag *mat.Dense, low, high float64) (*image.Gray, er
 	}
 	return edges, nil
 }
+
+// ImageToYCbCrForTesting converts an image to YCbCr. It is only to be used for testing.
+func ImageToYCbCrForTesting(dst *image.YCbCr, src image.Image) {
```
Comment (tessavitabile) [inline rimage/image_processing.go:513]: I couldn't find another was to convert to YCbCr, so I exported this function. Let me know if you have another idea!

File: services/slam/builtin/orbslam_int_test.go [hunk 2]
```diff
@@ -89,12 +103,22 @@ func TestOrbslamIntegration(t *testing.T) {
...
+
 	attrCfg := &builtin.AttrConfig{
 		Algorithm: "orbslamv3",
-		Sensors:   []string{"orbslam_int_color_camera", "orbslam_int_depth_camera"},
+		Sensors:   sensors,
 		ConfigParams: map[string]string{
-			"mode":              "rgbd",
-			"orb_n_features":    "1000",
+			"mode":              reflect.ValueOf(mode).String(),
+			"orb_n_features":    "1250",
```
Comment (tessavitabile) [inline services/slam/builtin/orbslam_int_test.go:121]: I decided to change `orb_n_features` to its default here.

File: rimage/image_processing.go (hunk 1 above)
Comment (bhaney) [reply rimage/image_processing.go:513]: you can probably use the draw.Draw function to do this
```
func MakeYCbCr(pic image.Image) *image.YCbCr {
     result := image.NewYCbCr(pic.Bounds())
    draw.Draw(result, result.Bounds(), pic, pic.Bounds().Min, draw.Src)
    return result
```

Comment (tessavitabile) [reply rimage/image_processing.go:513]: It doesn't look like I can do this, since image.YCbCr doesn't implement draw.Image (missing method Set).

Comment (bhaney) [reply rimage/image_processing.go:513]: oh wow, yea it looks like YCbCr isn't pixel addressable! So, yea, this function seems like the best bet for now

**Review by bhaney — APPROVED** [review]
LGTM, as an alternative, you can decode JPEG images directly to YCbCr rather than converting the PNG images to to YCbCr

File: rimage/image_processing.go (hunk 1 above)
Comment (bhaney) [reply rimage/image_processing.go:513]: Just found an alternative - but if you would like to stick to using one set of images (the png sources) then what you have now seems like the way to go

File: services/slam/builtin/builtin_test.go [hunk 3]
```diff
@@ -322,6 +351,44 @@ func setupInjectRobot() *inject.Robot {
...
+					if err != nil {
+						return nil, err
+					}
+					var ycbcrImg image.YCbCr
+					rimage.ImageToYCbCrForTesting(&ycbcrImg, img)
+					return gostream.NewEmbeddedVideoStreamFromReader(
+						gostream.VideoReaderFunc(func(ctx context.Context) (image.Image, func(), error) {
+							return &ycbcrImg, func() {}, nil
+						}),
+					), nil
```
[the remark this answers is not in the document]
Comment (bhaney) [reply services/slam/builtin/builtin_test.go:386]: But you need the libjpeg library to use that package - which may not be present on our CI/canon

**Review by kkufieta — APPROVED** [review]
LGTM

File: services/slam/builtin/builtin_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (tessavitabile) [reply services/slam/builtin/builtin_test.go:386]: Thanks for the idea! I think I'll leave as is.

**Pull request merged**